src/promptflow/promptflow/batch/_csharp_executor_proxy.py

```diff
@@ -23,115 +28,53 @@ def create(
         *,
         connections: Optional[dict] = None,
         storage: Optional[AbstractRunStorage] = None,
-    ) -> "AbstractExecutorProxy":
+        **kwargs,
+    ) -> "CSharpExecutorProxy":
         """Create a new executor"""
-        working_dir = working_dir or flow_file.parent
-
-        from pythonnet import load
-
-        # pythonnet.load must be called before import clr
-        load("coreclr")
-        import clr
-
-        with _change_working_dir(working_dir):
-            sys.path.append(working_dir.as_posix())
-            clr.AddReference("Promptflow")
-            sys.path.pop()
-            from Promptflow.Executor import YamlExecutor
-
-            connection_provider_url = ""
-            executor = YamlExecutor(
-                flow_file.read_text(encoding="utf-8"), connection_provider_url, working_dir.as_posix()
-            )
-            return cls(executor)
+        port = cls.find_available_port()
+        log_path = kwargs.get("log_path", "")
```

Comment:
why this is in kwargs? it's needed for csharp but not for python?

Reply:
Yes. We add kwargs to the batch engine to receive some custom parameters when creating different executor proxy. 

For example, runtime needs to pass `connection_endpoint`, `log_path`, `container_name...` to create C# executor proxy, but creating python executor proxy does not require these.